models_evaluation.py

```diff
@@ -18,41 +20,43 @@ def questions_reading(analogies):
     :param analogies: string
     :return: dict
     """
-    analogy_dict = dict()
+    analogy_dict = defaultdict(list)
     with open(analogies, "r") as analogy_f:
         key = None
-        for line in analogy_f:
-            if line in ['\n', '\r\n']:
+        for line in map(str.strip, analogy_f):
+            if not line.strip():
                 continue
             if line.startswith(":"):
                 # new section starts
                 key = line.rstrip()
-                analogy_dict[key] = list()
             else:
-                analogy_dict[key].append(line[:-1].split("\t"))
+                analogy_dict[key].append(line.split("\t"))
     return analogy_dict
 
 
-def analogy_evaluation(models_path, analogies, first_n):
+def analogy_evaluation(models_path, analogies, first_n, file_out):
     """
-    Compute performance of the model on an analogy test set.
+    Compute performance of the model on an analogy test set and writes results to a file.
     :param models_path: string
     :param analogies: dict
     :param first_n: int
-    :return: DataFrame
+    :param file_out: string
+    :return: None
     """
-    result = pd.DataFrame(
-        columns=["Model", "Type of questions", "Accuracy1 (for present questions)", "Accuracy2 (for all questions)",
-                 "Number of questions", "Number of present questions", "No words in vocabulary"])
-
-    for file in listdir(models_path):
-        if not isfile(join(models_path, file)):
+    csv_file = open(file_out, 'w')
+    writer = csv.writer(csv_file)
```

Comment:
Just in case, there is also csv.DictWriter, which is slightly better for dict-like objects. But you can safely ignore this comment